app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -84,15 +84,15 @@ def activity_session_params
                   :completed_at,
                   :activity_uid,
                   :anonymous,
-                  concept_tag_results_attributes:  concept_tag_keys)
+                  concept_results_attributes:  concept_tag_keys)
       .merge(data: @data).reject {|k,v| v.nil? }
   end
 
   # Grab a list of all the arbitrarily-named keys that are provided in the concept tag results payload.
   # Returns a list of symbols, e.g. [:student_input, :wpm]
-  def concept_tag_result_allowed_keys
-    if params[:concept_tag_results_attributes]
-      params[:concept_tag_results_attributes].reduce [] do |acc, hash|
+  def concept_result_allowed_keys
```

Comment:
I'm a bit confused - why are we determining what the keys happen to be, and then explicitly permitting only those keys? Isn't this the same as just not enforcing requirements on the keys?